### ISSUE TITLE
Why not just completely open source ESP Rainmaker? (MEGH-2878)

### ISSUE THREAD

**Issue by NemesisXB** [espressif/esp-rainmaker#96]
Why don't you just completely open source ESP Rainmaker, including the server side code? Espressif is after all a silicon vendor. The pricing is just too much for small to medium projects. Also, I have open issues from almost 18 months ago that require server side updates, which has not been done as only espressif can make those changes. If it was actually open source it might have been done by someone else. You can put a clause in the license or create a new license that states that it may only be used with espressif chips.

**Comment by Xarkle1** [comment]
I agree - open sourcing Rainmaker would be great.   We are developing IOT using ESP32/S2/S3/C3 and that would be super helpful and great for Espressif.

**Comment by felixcollins** [comment]
At the very least there should be some public information about expected costs of using it. One gets the impression it will be cheap or free for small commercial use. Now I'm worried that I'm wasting my time investigating it!

**Comment by halfn0rmal** [comment]
I have enquired about pricing and it is prohibitive for a small company, minimum 5000usd per year for usage plus 5000usd upfront for the first year. If your are a new venture and you are not sure whether to use rain maker then I would say you are better to invest that 10,000usd and just use the aws examples with amplify, iot core etc and build your own infrastructure that you can own going forward. For a larger company adding cloud to an existing product then this may make sense for you.
To save someone the time of enquiring the costs are below:
1) License (Espressif pays AWS bill) - our recommended model
In this model, if the amount of communication from the device is known, then Espressif can provide an option of one-time per device license fee and then Espressif pays the AWS bill directly. Please note that there is not much customisability for the cloud backend in this case and you get standard RainMaker deployment. If your device sends data to cloud 5 times a day, indicative license pricing is USD 0.40 per device.

2) Subscription (You pay your AWS bill)
In this model, you pay AWS bill directly and pay subscription fee to Espressif towards using ESP RainMaker. Annual recurring subscription fee is 0.10 USD per year per device (minimum of USD 5000 per year). You can also customise the RainMaker backend to add your own code (e.g. in this case to process the data).

Please note that there is one-time fee of USD 5000 that is applicable for both the models. RainMaker will be deployed in your own AWS account and you’ll get to choose which region of AWS to choose.

**Comment by felixcollins** [comment]
Okay! Thanks @halfn0rmal. I guess that kills that idea then. I posted on the forum asking about licensing and limitations of operating on the free platform.

**Comment by NemesisXB** [comment]
Another thing to note is that the pricing changed quite a bit since I enquired about it. My quote was that when deploying on your own AWS account, the upfront cost was $10k but no yearly subscription. I just can't see this working. Bigger companies that can afford this probably has the resources to implement their own, and for smaller companies that don't have the resources, these prices are exorbitant. It also completely takes startups and entrepreneurs out of the picture.

**Comment by halfn0rmal** [comment]
I would have thought it would be in espressifs interest to make it no upfront costs to onboard new customers to their hardware platform. It would be a good motivation to use their hardware.

For me I think it isn't worth it. There is a bit of work in rolling your own but long term it would pay off.

**Comment by felixcollins** [comment]
I agree with all of that. We actually have some of the cloud infrastructure already set up for ota, telemetry etc using free tier of firebase. The easy and secure provisioning is a major draw. At the moment we use the wifi local access point and a web page. A bit old school and complex for a lot of our customers.

**Comment by ameyinamdar** [comment]
@felixcollins you can use the Unified provisioning component (supports both SoftAP and BLE transports) with open-source phone apps. Please check out https://github.com/espressif/esp-idf/tree/master/examples/provisioning

**Comment by chaudhariatul** [comment]
Is there a table which shows list of boards that support SoftAP, BLE or both?

**Comment by shahpiyushv** [comment]
@chaudhariatul , basically all chips that support BLE can support both SoftAP as well as BLE provisioning viz. esp32, esp32c3, esp32s3, esp32c2. However, only one of these can be kept enabled at a time (using menuconfig options)

esp32s2 does not have BLE and so it can support only SoftAP provisioning.

**Comment by Bsm-B** [comment]
this topic is very interesting, but i think is best solution try to change endpoint to a private MQTT broker ,OTA via private server  and Give up some features Of course if he doesn't disagree with License of opensource rainmaker.

I think is the best way , i dont know if someone try to do this?  share with us

**Comment by dzungpv** [comment]
I think Espressif, if could not complete open the code, they can partly open the code (Community version), like there are some limitation set of advance feature and then charge for advance feature with fully version (Professional version) like many project. So developer can fix the bug and improve the platform.
